cfn-templates/data-exports-aggregation.yaml

```diff
@@ -0,0 +1,1091 @@
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Policies:
+        - PolicyName: "ExecutionDefault"
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+            - Effect: Allow
+              Action:
+              - logs:CreateLogStream
+              - logs:PutLogEvents
+              - logs:CreateLogGroup
+              Resource:
+              - !Sub "arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/${ResourcePrefix}-CID-CUR2Creator"
+              - !Sub "arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/${ResourcePrefix}-CID-CUR2Creator:*"
+              - !Sub "arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/${ResourcePrefix}-CID-CUR2Creator:*:*"
+        - PolicyName: "ExecutionSpecific"
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+            - Effect: Allow
+              Action:
+              - cur:PutReportDefinition
+              - bcm-data-exports:CreateExport
+              - bcm-data-exports:UpdateExport
+              - bcm-data-exports:DeleteExport
+              Resource:
+                - Fn::Sub: arn:${AWS::Partition}:bcm-data-exports:*:${AWS::AccountId}:export/*
+
+  CIDLambdaCUR2Creator:
+    Type: AWS::Lambda::Function
+    Condition: DeployCUR2ViaLambda
+    Properties:
+      Runtime: python3.11
+      FunctionName:
+        Fn::Sub: ${ResourcePrefix}-CID-CUR2Creator
+      Handler: index.lambda_handler
+      MemorySize: 128
+      Role:
+        Fn::GetAtt: CIDLambdaCUR2CreatorRole.Arn
+      Timeout: 15
+      Code:
+        ZipFile: |
+          import os
+          import json
+          import uuid
+
+          import boto3
+          import cfnresponse
+
+          region = os.environ['AWS_REGION']
+
+          # CUR2 only exists in us-east-1 and cn-northwest-1 regions
+          if region.startswith('cn-'):
+              region = 'cn-northwest-1'
+          else:
+              region = 'us-east-1'
+
+          client = boto3.client('cur', region_name=region)
+
+          def lambda_handler(event, context):
+
+            print(json.dumps(event))
+            reason = ""
+
+            try:
+              export = event['ResourceProperties']['Export']
+              export_name = event['ResourceProperties']['Export']['Name']
+
+              if event['RequestType'] == 'Create':
+                  res = client.put_export_definition(
+                      Export=export
+                  )
+                  print(json.dumps(res))
+              elif event['RequestType'] == 'Update':
+                  old_export_name = event['OldResourceProperties']['Export']['Name']
+                  if export["Name"] != old_export_name:
+                      res = client.create_export(
+                          Export=export
+                      )
+                      print(json.dumps(res))
```

Comment:
do you need to delete old one as well?

Reply:
in case when you change the name on update and you have to create the new CUR, you need to delete old. Correct ?